Library/Formula/llvm.rb

```diff
@@ -116,18 +152,66 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    if build.with?("lldb") && build.without?("clang")
-      raise "Building LLDB needs Clang support library."
+    if build.with? "clang"
+      (buildpath/"tools/clang").install resource("clang")
     end
 
-    if build.with? "clang"
+    if build.with? "extra-tools"
+      if build.with? "clang"
+        (buildpath/"tools/clang/tools/extra").install resource("extra-tools")
+      else
+        odie "clang's extra tools must be built with clang"
+      end
+    end
+
+    if build.with? "libcxx"
       (buildpath/"projects/libcxx").install resource("libcxx")
-      (buildpath/"tools/clang").install resource("clang")
-      (buildpath/"tools/clang/tools/extra").install resource("clang-tools-extra")
+      (buildpath/"projects/libcxxabi").install resource("libcxxabi")
```

Comment:
The "libcxxabi" bit is actually no-op due to build script of libcxx only link to system libc++abi. Also it makes more sense to actually link to system libc++abi, otherwise there may be potentially interoperability issue when upstream libc++abi is not 100% incompatible with the system one, which could lead to obscure errors at run-time.

The actual case that does need libc++abi is on Snow Leopard, which didn't ship with libc++abi. However I guess it is not officially supported by now.